Dialog/SpritesEditorDialog.cpp

```diff
@@ -0,0 +1,561 @@
+{
+    // Find if new entityset data exist
+    LevelComponents::EntitySet *curEntityset = GetCurEntitySetPtr(true);
+
+    // Generate Entityset Load Table and update them into the new entityset instance
+    QStringList loadtableStrData = ui->lineEdit_SpritesetLoadTable->text().split(QChar(' '), Qt::SkipEmptyParts);
+    if (loadtableStrData.size() & 1)
+    {
+        QMessageBox::critical(this,
+                              QString(QObject::tr("Error")),
+                              QString(QObject::tr("Illegal LoadTable element size!\nThere should be an even number of elements.")));
+        return;
+    }
+    if (loadtableStrData.size() > (0x1F * 2))
+    {
+        QMessageBox::critical(this,
+                              QString(QObject::tr("Error")),
+                              QString(QObject::tr("Too many elements. Max number of elements: 62."))); // 0x1F * 2 = 62
+        return;
+    }
+    for (int i = 0; i < loadtableStrData.size(); i += 2)
+    {
+        if (loadtableStrData[i].toUInt(nullptr, 16) > 0x80)
+        {
+            QMessageBox::critical(this,
+                                  QString(QObject::tr("Error")),
+                                  QString(QObject::tr("Illegal entity Id at element %1, it should be less than 0x81.")).arg(i));
+            return;
+        }
+        if (loadtableStrData[i + 1].toUInt(nullptr, 16) > 7)
+        {
+            QMessageBox::critical(this,
+                                  QString(QObject::tr("Error")),
+                                  QString(QObject::tr("Illegal palette offset Id at element %1, it should be less than 8.")).arg(i + 1));
+            return;
+        }
+    }
+    curEntityset->ClearEntityLoadTable();
+    for (int i = 0; i < loadtableStrData.size(); i += 2)
+    {
+        LevelComponents::EntitySetinfoTableElement newelement;
+        newelement.Global_EntityID = loadtableStrData[i].toUInt(nullptr, 16);
+        newelement.paletteOffset = loadtableStrData[i + 1].toUInt(nullptr, 16);
+        curEntityset->EntityLoadTablePushBack(newelement);
+    }
+
+    // UI update
+    RenderSpritesetTileMapAndResetLoadTable();
+}
+
+/// <summary>
+/// Export sprite tiles map to file when hit pushButton_SpriteTilesExport
+/// </summary>
+void SpritesEditorDialog::on_pushButton_SpriteTilesExport_clicked()
+{
+    QString qFilePath = QFileDialog::getSaveFileName(this, tr("Save current Tile8x8 map to a file"),
+                                                     QString(""), tr("PNG file (*.png)"));
+    if (qFilePath.compare(""))
+    {
+        LevelComponents::Entity *curEntity = GetCurEntityPtr();
+        int tileNum = curEntity->GetTilesNum();
+        QPixmap SpriteTilesPixmap(8 * 32, (tileNum / 32) * 8);
+        QPainter SpriteTilemapPixmapPainter(&SpriteTilesPixmap);
+        SpriteTilemapPixmapPainter.drawImage(0, 0, FileIOUtils::SetBGColor(curEntity->GetTileMap(curEntityPalId), this));
+        SpriteTilesPixmap.save(qFilePath, "PNG", 100);
```

Comment:
let's do lowercase png

Reply:
no, that's some defined things in Qt, the file type is lowcased as a result.